2.evaluate_data/nbconverted/predict_sc_probabilities.py

```diff
@@ -0,0 +1,185 @@
+# In this part of the analysis, the single cell probabilities are predicted and stored in a parquet file.
+# The index of this file is used to reference the single cell index in the corresponding parquet files, where the "Metadata_plate" column refers to the plate name corresponding to the reference file.
+
+# ### Import Libraries
+
+# In[1]:
+
+
+import gzip
+import io
+import pathlib
+
+import pandas as pd
+import requests
+from joblib import load
+
+
+# ## Find the path of the git directory
+
+# In[2]:
+
+
+# Get the current working directory
+cwd = pathlib.Path.cwd()
+
+if (cwd / ".git").is_dir():
+    root_dir = cwd
+
+else:
+    root_dir = None
+    for parent in cwd.parents:
+        if (parent / ".git").is_dir():
+            root_dir = parent
+            break
+
+# Check if a Git root directory was found
+if root_dir is None:
+    raise FileNotFoundError("No Git root directory found.")
+
+
+# ## Load data code
+
+# In[3]:
+
+
+def load_joblib_from_url(url):
+    """
+    Retirieve joblib or gzip csv file from url
+
+    Parameters
+    ----------
+    url : url
+        The raw url of the file
+
+    Returns
+    -------
+    obj : any Python object
+    """
+
+    response = requests.get(url)
+    content = response.content
+    file_object = io.BytesIO(content)
+
+    if ".csv" in url:
+        file_object = gzip.GzipFile(fileobj=file_object)
+        obj = pd.read_csv(file_object)
+
+    elif ".joblib" in url:
+        obj = load(file_object)
+
+    return obj
+
+
+# In[4]:
+
+
+# The paths of the models
+model_paths = ["https://github.com/WayScience/phenotypic_profiling_model/raw/main/2.train_model/models/multi_class_models/final__CP.joblib", "https://github.com/WayScience/phenotypic_profiling_model/raw/main/2.train_model/models/multi_class_models/shuffled_baseline__CP.joblib"]
+
+# The path of the data used for the data used for inferencing in the phenotypic_profiling_model repo
+data_path = "https://github.com/WayScience/phenotypic_profiling_model/raw/main/0.download_data/data/labeled_data.csv.gz"
+
+# Path to the drive
+drive_path = f"{root_dir}/big_drive"
+
+# The predicted probabilities of the models on each cell from each plate
+output_proba_path = f"{drive_path}/probability_sc_data"
+
+# The path of the normalized sc data
+norm_data_path = pathlib.Path(f"{drive_path}/normalized_sc_data")
+
+# The path of the model's predicted probabilities
+pathlib.Path(f"{output_proba_path}").mkdir(parents=True, exist_ok=True)
+
+
+# ## Load the data
+
+# In[5]:
+
+
+# Store the models as a dictionary
+models = {model_path.split("/")[-1].split("__")[0]: load_joblib_from_url(model_path) for model_path in model_paths}
+
+# Original dataset used to select features for models
+data = load_joblib_from_url(data_path)
+
+# Data columns from original dataset
+all_cols = data.columns
+
+
+# ## Inference
+
+# In[6]:
+
+
+# Extract CP features from all columns depending on desired dataset
+feature_cols = [col for col in all_cols if "CP__" in col]
```

Comment:
```suggestion


# ## Inference

# In[6]:


# Extract CP features from all columns depending on desired dataset
feature_cols = [col for col in data.columns if "CP__" in col]
```
seems like `all_cols` was only used once. Use `data.columns` inside the list comprehension to avoid using extra memory space.

Reply:
That a nice small simplification, changed!